Adapter_Revit_UI/AdapterActions/Push.cs

```diff
@@ -40,19 +41,44 @@ public partial class RevitUIAdapter
 
         public override List<object> Push(IEnumerable<object> objects, string tag = "", PushType pushType = PushType.AdapterDefault, ActionConfig actionConfig = null)
         {
-            bool success = true;
+            // Check the document
+            UIDocument uiDocument = this.UIDocument;
+            Document document = this.Document;
+            if (document == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("BHoM objects could not be removed because Revit Document is null.");
+                return new List<object>();
+            }
 
-            // ----------------------------------------//
-            //                 SET-UP                  //
-            // ----------------------------------------//
+            if (document.IsReadOnly)
+            {
+                BH.Engine.Reflection.Compute.RecordError("BHoM objects could not be removed because Revit Document is read only.");
+                return new List<object>();
+            }
 
+            if (document.IsModifiable)
+            {
+                BH.Engine.Reflection.Compute.RecordError("BHoM objects could not be removed because another transaction is open in Revit.");
+                return new List<object>();
+            }
+
+            // Set config
+            RevitPushConfig pushConfig = actionConfig as RevitPushConfig;
+            if (pushConfig == null)
+            {
+                BH.Engine.Reflection.Compute.RecordWarning("Revit Push Config has not been specified. Default Revit Push Config is used.");
+                pushConfig = RevitPushConfig.Default;
+            }
+
+            // Suppress warnings
+            if (UIControlledApplication != null && pushConfig.SuppressFailureMessages)
+                UIControlledApplication.ControlledApplication.FailuresProcessing += ControlledApplication_FailuresProcessing;
```

Comment:
Maybe worth adding a `RecordWarning` or `RecordMessage` to kindly remind the user that no other warning will appear because they're hidden? 😄 Not sure actually, just thought if I receive a script from somebody else with "suppress all Warning" turned on I'd want to make sure I realise that they are suppressed

Reply:
Hmm not sure if I like the idea - this might easily lead to the case where we will need to warn the user about multiple settings that he/she set up on purpose. If the warnings were suppressed by default, I would agree about leaving the warning, but why would we warn the user about his/her own actions?